tests/mmm/test_transformers.py

```diff
@@ -65,8 +67,29 @@ def test_batched_convolution(convolution_inputs, convolution_axis):
     x_val = np.moveaxis(
         x_val if x_val is not None else getattr(x, "value", x), convolution_axis, 0
     )
-    assert np.allclose(y_val[0], x_val[0])
-    assert np.allclose(y_val[1:], x_val[1:] + x_val[:-1])
+    mode_assertions = {
+        ConvMode.Before: lambda: (
+            np.allclose(y_val[0], x_val[0]),
+            np.allclose(y_val[1:], x_val[1:] + x_val[:-1]),
+        ),
+        ConvMode.After: lambda: (
+            np.allclose(y_val[-1], x_val[-1]),
+            np.allclose(y_val[:-1], x_val[1:] + x_val[:-1]),
+        ),
+        ConvMode.Overlap: lambda: (
+            np.allclose(y_val[0], x_val[0]),
+            np.allclose(y_val[1:-1], x_val[1:-1] + x_val[:-2]),
+        ),
+    }
+
+    assert all(mode_assertions[mode]())
```

Comment:
Nitpick, use if else instead of dict and use `np.testing.assert_allclose` which has a nicer error message when the two don't match.

```suggestion
    mode_assertions = {
        ConvMode.Before: lambda: (
            np.allclose(y_val[0], x_val[0]),
            np.allclose(y_val[1:], x_val[1:] + x_val[:-1]),
        ),
        ConvMode.After: lambda: (
            np.allclose(y_val[-1], x_val[-1]),
            np.allclose(y_val[:-1], x_val[1:] + x_val[:-1]),
        ),
        ConvMode.Overlap: lambda: (
            np.allclose(y_val[0], x_val[0]),
            np.allclose(y_val[1:-1], x_val[1:-1] + x_val[:-2]),
        ),
    }

    assert all(mode_assertions[mode]())
```